awslambda/authorizer/index.js

```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+console.log('Loading function')
+
+const aws = require('aws-sdk')
+const axios = require('axios')
+
+const httpsOpts = (token) => ({
+  headers: {
+    'Authorization': `token ${token}`,
+  }
+})
+
+const generatePolicy = (principleId, effect, resource) => {
+  let authResponse = {};
+
+  authResponse.principleId = principleId;
+  if (effect && resource) {
+    const policyDocument = {
+      Version: '2012-10-17', //Default Version
+      Statement: [{
+        Action: 'execute-api:Invoke', //Default Action
+        Effect: effect,
+        Resource: resource
+      }]
+    };
+    authResponse.policyDocument = policyDocument;
+  }
+  return authResponse;
+}
+
+exports.handler = (event, context, callback) => {
+  const token = event.authorizationToken;
+  const opts = httpsOpts(token);
+  axios.get('https://api.github.com/', opts)
+    .then(() => {
+      callback(null, generatePolicy('user', 'Allow', event.methodArn))
+    })
+    .catch(() => {
+      callback('Unauthorized')
+    })
```

Comment:
From the Google article.
> You can also use catch():
> 
> `get('story.json').then(function(response) {
>   console.log("Success!", response);
> }).catch(function(error) {
>   console.log("Failed!", error);
> })`
> 
> There's nothing special about catch(), it's just sugar for then(undefined, func), but it's more readable. Note that the two code examples above do not behave the same, the latter is equivalent to:
> 
> `get('story.json').then(function(response) {
>   console.log("Success!", response);
> }).then(undefined, function(error) {
>   console.log("Failed!", error);
> })`
> 
Nothing incorrect about it. Just a way of separating success and failure.